tmail-backend/guice/distributed/src/main/java/com/linagora/tmail/ScheduledReconnectionHandler.java

```diff
@@ -339,16 +349,24 @@ private Mono<Void> restart() {
                 .then());
     }
     
-    private boolean restartNeeded() {
+    public boolean restartNeeded() {
         return QUEUES_TO_MONITOR.stream()
-            .filter(this::hasNoConsumers)
-            .findAny()
-            .isPresent();
+            .anyMatch(queue -> !hasConsumers(queue));
     }
 
-    private boolean hasNoConsumers(String queue) {
-        return mqManagementAPI.queueDetails(configuration.getVhost().orElse(DEFAULT_VHOST), queue)
-            .getConsumerDetails()
-            .isEmpty();
+    private boolean hasConsumers(String queue) {
+        try {
+            boolean hasConsumers = !mqManagementAPI.queueDetails(configuration.getVhost().orElse(DEFAULT_VHOST), queue)
+                .getConsumerDetails()
+                .isEmpty();
+
+            if (!hasConsumers) {
+                LOGGER.warn("The {} queue has no consumers", queue);
+            }
+
+            return hasConsumers;
+        } catch (RabbitMQManagementAPI.QueueNotFoundException e) {
+            return true;
```

Comment:
false?

Reply:
No, it should be `true`. not existing queue would be considered `hasConsumers` so the restart won't be triggered all the time.

Reply:
I don't have an issue with the logic (it might be easier with if-else or filter). 

Just the naming is confusing; when the queue is not found => "has consumers." @@

Reply:
Ok I will try to improve readbility here

Reply:
should be better now